dapp/src/auctions/hooks/useAuctions.ts

```diff
@@ -76,39 +85,69 @@ export function useAuctions({
     const indexerClient = useIotaNamesIndexerClientContext();
     const { data: auctionHouseData } = useAuctionHouse();
 
-    // Determine the appropriate query key and fetch function based on filter
-    const queryKeyBase = userAddress
-        ? queryKey.userAuctionHistory(userAddress)
-        : queryKey.auctionList();
+    const shouldFetchAllAuctions = type === 'allAuctions';
+    const shouldFetchUserAuctions =
+        !!userAddress && (type === 'userAuctions' || type === 'allAuctions');
+
+    const {
+        data: allAuctions = NAMES_PLACEHOLDER,
+        isLoading: isLoadingAllAuctions,
+        error: allAuctionsError,
+    } = useQuery<AuctionsResponse>({
+        // eslint-disable-next-line @tanstack/query/exhaustive-deps
+        queryKey: [...queryKey.auctionList(), search, status, sort, sortBy, page, pageSize, type],
```

Comment:
Works the same but makes more sense
```suggestion
        queryKey: [...queryKey.auctionList(), search, status, sort, sortBy, page, pageSize, shouldFetchAllAuctions],
```